### PR TITLE
Adding minor updates as examples to healthcare.gov package for header and footer

### PR DESCRIPTION
**Pull request by niralkunadia12**
Adding minor updates as examples to healthcare.gov package for header  bottom and footer top section

[LHDM-262](https://jira.cms.gov/browse/LHDM-262): Contribute minor update to `<Header/>` and `<Footer/>` back to DS

## Summary
Added Examples to the [healthcare.gov](https://cmsgov.github.io/design-system/healthcare/) for header and footer.
Select `Patterns` from left menu, Choose header or footer to see changes with examples.

### Added
Header
Patterns->Header->Added example for Header bottom section.(Header - Bottom)
Footer
Patterns->Footer->Added example for Footer top section.(Footer - Top Section)

Please find attached images to compare changes.

Footer Before
Footer_after

Footer After
Footer_before

Header Before
Header_before

Header After
Header_after
r

### CODE REVIEW

File: packages/ds-healthcare-gov/docs/src/pages/patterns/Footer/Footer.example.jsx [hunk 1]
```diff
@@ -45,12 +45,27 @@ class FooterExample extends React.PureComponent {
...
-        <Footer initialLanguage={this.locale} />
-        {this.renderFooterToggles()}
+    const wrapperClassNames = 'ds-u-padding--1';
+    const footerTopExample = (
+      <div className="ds-l-container ds-u-padding-bottom--2">
+        <div className="ds-u-text-align--right ds-u-border-bottom--1 ds-u-padding-bottom--2 ds-u-padding-right--2">
+          <span className="ds-u-font-size--xl ds-u-sans ds-u-font-weight--bold ds-u-color--gray-dark">
+            58 DAYS
+          </span>
+        </div>
```
Comment (DavidMCarek) [inline packages/ds-healthcare-gov/docs/src/pages/patterns/Footer/Footer.example.jsx:55]: Super small nit pick, we could probably simplify this down to fewer nested elements.

**Review by DavidMCarek — COMMENTED** [review]
Everything looks pretty solid but I noticed that the build was failing. I'm not really sure why though,

**Comment by niralkunadia12** [comment]
> Everything looks pretty solid but I noticed that the build was failing. I'm not really sure why though,

Did you run this : `yarn update-snapshots` ?

**Comment by DavidMCarek** [comment]
I mean the ci build was failing. It looks like it might have been some weird network issue.

File: packages/ds-healthcare-gov/docs/src/pages/patterns/Footer/Footer.example.jsx [hunk 2]
```diff
@@ -45,12 +45,25 @@ class FooterExample extends React.PureComponent {
...
       </div>
     );
+    return (
+      <>
+        <div className="ds-u-margin-top--4">
+          <h6 className="preview__label">Minimal</h6>
+          <Footer initialLanguage={this.locale} />
+          <h6 className="preview__label">Footer - Top Section</h6>
+          <Footer initialLanguage={this.locale} footerTop={footerTopExample} />
+          <div className={wrapperClassNames}>{this.renderFooterToggles()}</div>
```
Comment (DavidMCarek) [inline packages/ds-healthcare-gov/docs/src/pages/patterns/Footer/Footer.example.jsx:63]: Any reason you didn't just set the class directly ie `className="ds-u-padding--1"`?

**Review by DavidMCarek — APPROVED** [review]
Everything looks good! Just had one final nit pick that's not worth blocking an approval for.

File: packages/ds-healthcare-gov/src/components/Footer/Footer.jsx [hunk 3]
```diff
@@ -49,6 +50,10 @@ _Footer.propTypes = {
    * `https://healthcare.gov/topics`.
    */
   primaryDomain: PropTypes.string,
+  /**
+   * Element to be added to display content for top footer section changes
+   */
+  footerTop: PropTypes.node,
```
Comment (scheul93) [inline packages/ds-healthcare-gov/src/components/Footer/Footer.jsx:56]: We've got some intermediate Typescript typing going on right now, so any time a prop is added, we also have to update the type definition file. Could you please update [this file](https://github.com/CMSgov/design-system/blob/ff52e04552042f02931fc93359325b93616e4f1d/packages/ds-healthcare-gov/src/types/Footer/index.d.ts#L6) with

```
  footerTop?: React.ReactNode;
```
in the `FooterProps` interface

File: packages/ds-healthcare-gov/src/components/Header/Header.jsx [hunk 4]
```diff
@@ -247,6 +248,10 @@ _Header.propTypes = {
    * the `defaultMenuLinks` method or the links provided by the `links` prop.
    */
   submenuBottom: PropTypes.node,
+  /**
+   * Element added to display content on Header bottom section
+   */
+  headerBottom: PropTypes.node,
```
Comment (scheul93) [inline packages/ds-healthcare-gov/src/components/Header/Header.jsx:254]: We've got some intermediate Typescript typing going on right now, so any time a prop is added, we also have to update the type definition file. Could you please update [this file](https://github.com/CMSgov/design-system/blob/ff52e04552042f02931fc93359325b93616e4f1d/packages/ds-healthcare-gov/src/types/Header/index.d.ts) with

```
  headerBottom?: React.ReactNode;
```
in the `HeaderProps` interface

File: packages/ds-healthcare-gov/docs/src/pages/patterns/Footer/Footer.example.jsx (hunk 2 above)
Comment (niralkunadia12) [reply packages/ds-healthcare-gov/docs/src/pages/patterns/Footer/Footer.example.jsx:63]: > 

There is no particular reason for using WrapperClassName, I found this in one of the DS file coding, So just thought to use same pattern.

File: packages/ds-healthcare-gov/src/components/Footer/Footer.jsx (hunk 3 above)
Comment (niralkunadia12) [reply packages/ds-healthcare-gov/src/components/Footer/Footer.jsx:56]: > 

@scheul93 , After adding `footerTop?: React.ReactNode;` line, in `index.d.ts`, I am getting Enlint error. Which is not letting me commit my changes. Any ideas?

```
✖ eslint found some errors. Please fix them and try committing again.
/Users/niralkunadia/Localdev/cms/design-system/packages/ds-healthcare-gov/src/types/Footer/index.d.ts
11:24  error  Parsing error: Unexpected token, expected "{"
 9 | 
10 | export class Footer extends React.Component<FooterProps, any> {
> 11 |   render(): JSX.Element;
   |                        ^
12 | }
```

Comment (niralkunadia12) [reply packages/ds-healthcare-gov/src/components/Footer/Footer.jsx:56]: @scheul93 , With this changes, Do I need to update snapshot? I am getting eslint error, for adding this line `footerTop?: React.ReactNode;`

Comment (niralkunadia12) [reply packages/ds-healthcare-gov/src/components/Footer/Footer.jsx:56]: Added `footerTop?: React.ReactNode;` to index.d.ts

File: packages/ds-healthcare-gov/src/components/Header/Header.jsx (hunk 4 above)
Comment (niralkunadia12) [reply packages/ds-healthcare-gov/src/components/Header/Header.jsx:254]: Done

**Review by pwolfert — APPROVED** [review]
LGTM 👍 Thanks for contributing these

**Comment by niralkunadia12** [comment]
Thank you all!

**Comment by pwolfert** [comment]
For some reason the CI job is treating this pull request differently (probably because it's a fork). It successfully ran the tests as a "pull request" under the multibranch pipeline job, but it's still waiting for it to execute as a "branch". I'm going to force this to be merged.

**Pull request merged**